futures/future_test.go

```diff
@@ -10,6 +10,18 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func Example() {
```

Comment:
what's the purpose this in the test,? doesn't seem to be used?

Reply:
This is not a test, it's an example. It will show up in the generated godocs..